Lib/asyncio/futures.py

```diff
@@ -173,7 +175,7 @@ def result(self):
         the future is done and has an exception set, this exception is raised.
         """
         if self._state == _CANCELLED:
-            raise exceptions.CancelledError
+            raise _create_cancelled_error(self._cancel_message)
```

Comment:
Why not simply `raise CancelledError(self._cancel_message)`?

Reply:
It's so that when no `msg` argument is provided (`self._cancel_message = None`), the exception object will be `CancelledError()` rather than `CancelledError(None)`.

This preserves the current behavior of cleaner tracebacks in this case by showing this for `traceback.print_exc()`:

```
  ...
  File "/.../cpython/Lib/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
asyncio.exceptions.CancelledError
```

instead of:

```
  ...
  File "/.../cpython/Lib/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
asyncio.exceptions.CancelledError: None
```

I also have tests to check this case.

Reply:
Maybe then 

```python
raise exceptions.CancelledError(
     '' if self._cancel_message is None else self._cancel_message)
```
?

I don't usually like exception constructors because they lengthen the traceback, often unnecessarily.

Reply:
Or make `msg=''` the default.

Reply:
I like the first one:

```python
raise exceptions.CancelledError(
     '' if self._cancel_message is None else self._cancel_message)
```